dali/operators/reader/loader/file_label_loader.cc

```diff
@@ -69,7 +69,7 @@ void FileLabelLoaderBase<checkpointing_supported>::ReadSample(ImageLabelWrapper
   });
   Index file_size = current_file->Size();
 
-  if (copy_read_data_ || !current_file->CanMemoryMap()) {
```

Comment:
Can't `CanMemoryMap` return false even uf copy_read_data_ is `true`? What about `s3://` storage? Quick scan of the code didn't reveal any tests for it, so we might silently break it, can't we?

Reply:
`copy_read_data_ = dont_use_mmap_ || !mmap_reserver_.CanShareMappedData()` - https://github.com/NVIDIA/DALI/blob/main/dali/operators/reader/loader/file_label_loader.h#L137

The idea is that `copy_read_data_` should reflect `CanShareMappedData` value I don't think we can break `s3` as it doesn't use/shouldn't use mmap. @jantonguirao ?

Reply:
I think that obtaining `CanMemoryMap` on a per-file basis is safer than inferring it somehow from `CanShareMappedData`.

Reply:
I think `CanShareMappedData` is a stronger check as it verifies if we can actually mmap all the files we want at the same time. This is important for single files as we can have many of them opened at a time. `CanMemoryMap` is whether a given object class can do it, but not necessarily if the OS has enough free FDs.

Reply:
Correct, the S3 code path won't do anything related to mmapping

Reply:
I agree with Michal, it is required to check on a per-file basis, because even if your system allows mmaping, if you get an S3 location, you can't call Get on it (it throws)

Reply:
Ok. If we can mix plain files and S3 objects inside one reader this makes a lot of sense. Removed.